tests/base.py

```diff
@@ -105,7 +105,7 @@ def expected_metadata(self):  # DOCS_BUG https://stitchdata.atlassian.net/browse
             "deals": {
                 self.PRIMARY_KEYS: {"dealId"},
                 self.REPLICATION_METHOD: self.INCREMENTAL,
-                self.REPLICATION_KEYS: {"hs_lastmodifieddate"},
+                self.REPLICATION_KEYS: {"property_hs_lastmodifieddate"},
```

Comment:
Should this include the `property_` prefix from the denesting method? None of the other replicaiton keys have this. And I think it will break every customer's connection that is replicating this stream.

Reply:
Currently, `hs_lastmodifieddate` is the bookmark key for the `deals` stream. `hs_lastmodifieddate` is available in the properties field at the nested level. 
For example, the response structure:
```
{
    "dealId": 1,
    "properties": {
        "hs_lastmodifieddate": 1653590080602
    }
}
```

As `hs_lastmodifieddate` is not available at the 1st level, it can not be marked as automatic inclusion and we can not select that particular field even. Also, tap includes all nested fields of the properties field as custom fields in the schema by appending the prefix `property_` along with each field.

For example, the schema of the deals stream looks like as below,
```
{
    "dealId": {"type": ["null", "integer"]},
    "properties": {"type": "object", "properties": {
            "hs_lastmodifieddate": {},
            "hs_likelihood_to_close": {}
        }
    },
    "property_hs_lastmodifieddate": {},
    "property_hs_likelihood_to_close": {}
}
```
That's why we are just marking  `property_hs_lastmodifieddate` as the inclusion of automatic. So, I am not sure in which case it will break the customer's connection because we have not removed or updated any fields.

Reply:
I believe this nested structure is true of the other streams as well, but `companies` for example denests the key and does not prefix `property_` onto the key. I think it will break downstream for customers because the value of the replication key is changing.

Reply:
@kspeer825  We have changed the bookmark key just for the `deals `stream. This change will not affect any other streams. 
If bookmark(`last_modifieddate`) is already available for the deals stream, then tap will use only that value for the first time. (#196 )
After completing sync, the tap will write an updated bookmark in the new bookmark key(i.e. `property_lastmodified_date`).
We haven't removed the existing bookmark as well, so in case we need to revert back, we can able to. 
After sync, the bookmark will look like the below.

```
  "bookmarks": {
        "stream_id_1": {
            "old_bookmark": "2022-04-08T00:00:00",
            "current_bookmark": "2022-04-15T00:00:00"
        }
    }
```

We have incorporated this change to fix the customer's issue. So, @KrisPersonal  shall we plan an alpha release first for the particular customer?